scripts/partition_manager_report.py

```diff
@@ -41,13 +41,23 @@ def print_region(region, size, pm_config):
     sorted_pm_config = sorted(sorted_pm_config, key=lambda x: pm_config[x]['address'])
 
     # Create text lines
-    lines = ['{}{}: {} ({}){}'.format(
+    lines = []
+    endspan = 0
+    for name in sorted_pm_config:
+        lines.append('{}{}: {} ({}){}'.format(
                 '| '+bcolors['WARNING'] if 'span' not in pm_config[name] else '+---'+bcolors['OKBLUE'],
                 hex(pm_config[name]['address']),
                 name,
                 hex(pm_config[name]['size']),
-                bcolors['ENDC'])
-            for name in sorted_pm_config]
+                bcolors['ENDC']))
+        if name == sorted_pm_config[:-1]:
+            continue
```

Comment:
This code will never work, so I suggest you remove it (or correct it into what it should do).

`name` is a single entry in `sorted_pm_config`, as example `mcuboot`.
`sorted_pm_config[:-1]` is an array, like:
```
['mcuboot', 'EMPTY_0', 'mcuboot_primary', 'mcuboot_pad', 'mcuboot_primary_app', 'spm', 'app', 'mcuboot_secondary', 'EMPTY_1']
```
So unless this is a special case handling, where array length is exactly two (resulting `sorted_pm_config[:-1]` tp become a single entry), then you'll never enter the `if,
So to my knowledge, `continue` will never be exectuted.

If you are handling such special case with array length == 2, then maybe just check for the length in that case.

Reply:
I meant to skip the last item in the array since a dottet line will be inserted below. I will change [:-1] to [-1]